File: del_without_headpointer.java
// Time Complexity : O(1) 
// Space Complexity :O(1)


public class del_without_headpointer {
   

public class DeleteWithoutHeadPointer {

	void deleteNode(ListNode del)
    {
        if(del.next ==null)
        del = null;
        else{
           del.val = del.next.val;
           del.next = del.next.next;
        }
    }

}

}

File: DelWithoutHeadPointer.py
# Time complexity is O(1)
# Space complexity is O(1)


class Solution:
    # Function to delete a node without any reference to head pointer.
    def deleteNode(self, curr_node):
        temp = curr_node.next
        curr_node.data = temp.data
        curr_node.next = temp.next
        temp.next = None
